File: 2350-shortest-impossible-sequence-of-rolls/2350-shortest-impossible-sequence-of-rolls.py
class Solution:
    def shortestSequence(self, rolls: List[int], k: int) -> int:
        s = set()
        res = 1
        for i in rolls:
            s.add(i)
            if len(s) == k:
                s.clear()
                res += 1
        return res

File: 2350-shortest-impossible-sequence-of-rolls/2350-shortest-impossible-sequence-of-rolls.cpp
class Solution {
public:
    int shortestSequence(vector<int>& rolls, int k) {
        unordered_set<int> s;
        int res = 1;
        for(int i : rolls) {
            s.insert(i);
            if(s.size() == k) {
                s.clear();
                ++res;
            }
        }
        return res;
    }
};